src/util/animation/AnimationBase.ts

```diff
@@ -17,42 +17,66 @@ const defaultAbort = () => false;
 export abstract class AnimationBase<
   T extends number | number[] = number | number[]
 > {
+  /**
+   * @see TAnimationBaseOptions
+   */
   readonly startValue: T;
   readonly byValue: T;
   readonly endValue: T;
   readonly duration: number;
   readonly delay: number;
   protected readonly easing: TEasingFunction<T>;
+  /**
+   * @private
+   * @see TAnimationCallbacks
+   */
   private readonly _onStart: VoidFunction;
   private readonly _onChange: TOnAnimationChangeCallback<T, void>;
   private readonly _onComplete: TOnAnimationChangeCallback<T, void>;
   private readonly _abort: TAbortCallback<T>;
   /**
-   * used to register the animation to a target object so it can be cancelled within hte object context
+   * Used to register the animation to a target object
+   * so that it can be cancelled within the object context
    */
   readonly target?: unknown;
 
   private _state: AnimationState = 'pending';
   /**
-   * time %
+   * Time %, or the ratio of `timeElapsed / duration`
+   * @see tick
    */
   durationRatio = 0;
   /**
-   * value %
+   * Value %, of the ratio of `(currentValue - startValue) / (endValue - startValue)`
    */
   valueRatio = 0;
   /**
-   * current value
+   * Current value
    */
   value: T;
   /**
-   * animation start time ms
+   * Animation start time ms
    */
   private startTime!: number;
 
   /**
-   * since both `byValue` and `endValue` are accepted in subclass options and are populated with defaults if missing,
-   * we defer to `byValue` and ignore `endValue` to avoid conflict
+   * Constructor
+   * Since both `byValue` and `endValue` are accepted in subclass options
+   * and are populated with defaults if missing, we defer to `byValue` and
+   * ignore `endValue` to avoid conflict
+   * @see TAnimationBaseOptions
+   * @see TAnimationValues
+   * @see TAnimationCallbacks
+   * @param startValue
+   * @param byValue
+   * @param duration
+   * @param delay
+   * @param easing
+   * @param onStart
+   * @param onChange
+   * @param onComplete
+   * @param abort
+   * @param target
```

Comment:
These `@param` directives are incorrect
You can do `@param {TAnimationBaseOptions} arg0` I think but I feell it is pointless. I don't mind

Reply:
What do you mean they are incorrect? The types are optional on javadoc param directives, and since we use typescript I don't see a point to include them

Reply:
We haven't looked into the pipeline that will generate docs to the website yet so we don't know if we will need JSDOC or not.
Anyways there are cases we do document essentials in JSDOC.
This isn't the case.
The syntax is wrong.
```ts
* @param {MyType} options
* @param [number] options.delay
```

No need for here.

Reply:
I'm sorry but I'm still confused. Do I need to change something? My understanding is that the types for the param directives are optional, and since the code has the typings it shouldn't be necessary. Is this incorrect?

Reply:
I will push the fix

Reply:
For me the important thing is the autocomplete/discovery docs VSCODE can offer.

https://user-images.githubusercontent.com/34343793/206888412-615a08bb-6a22-410c-83a6-afa254e5003e.mp4

Since it is able to extract comments from types seems to me it makes a lot of JSDOC redundant.
I prefer types being documented where possible instead of signatures, but there are signatures that must be typed for clarity as well. It is not strict, which is good.

We are still in search of the best practice so it remains to be resolved in the future.

Reply:
a89ce3592